src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java

```diff
@@ -158,6 +158,7 @@ private static String encloseInGroup(String expression) {
      * @return a regex string.
      */
     private static String toRegex(String input) {
+        // noinspection DynamicRegexReplaceableByCompiledPattern
```

Comment:
it looks like that more common way is to do like
`@noinspection UseOfPropertiesAsHashtable` in javadocs

Reply:
```
/**
     * Converts a normal package name into a regex pattern by escaping all
     * special characters that may occur in a java package name.
     *
     * @param input the input string.
     * @return a regex string.
     */
    @SuppressWarnings("DynamicRegexReplaceableByCompiledPattern")
    private static String toRegex(String input) {
        return input.replace(DOT, DOT_REGEX);
    }
```
If you want to suppress for entire method, IntelliJ is suggesting this.

Reply:
```// noinspection DynamicRegexReplaceableByCompiledPattern``` suppresses it for one statement only

Reply:
ok, lets keep it small

Reply:
@shashwatj07
Please move it to javadoc, we do not need to polite code with  comments

Reply:
done

Reply:
```@noinspection DynamicRegexReplaceableByCompiledPattern``` didn't seem to work in my IDE. Please see.

If it's not working we can do this https://github.com/checkstyle/checkstyle/pull/10103#discussion_r649042275